### DISCUSSION TITLE
Concerning my request for a Unimarc and 995 import.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#156] [General]
concerning my request for an unimarc + 995 import, I don't know if you've looked at catmandu but if you're looking for a marc / unimarc parser : catmandu is for you :-)

I don't know if it's easy to integrate (perl in c++ qt) but I think it does the job and should be integrated like the yaz tools

If that is impossible I will keep the independent version of catmandu in script but not being a programmer, I still managed to parse the unimarc and make the csv in 1 week (it's still not perfect because I'm not familiar with regular expressions) and the fix language is already quite complicated for me.

The most missing thing is some fields with customizable title in BiblioteQ in the books tables to accommodate the unimarc records info.

**Comment by textbrowser** [comment]
I have walls to paint now so I cannot look at it. I will see if an improvement can be made regarding the other items (dates, conversions of ISBNs in import, etc.) later. Perhaps another day. :)

**Reply by meteos77** [reply]
I hope your painting is going well (and that BiblioteQ and especially meteos77 don't take too much time :-)

I do it in spring, I can open the windows to air (the climate is around 0 ; 5° C (32 ; 45° F in my area at the moment - snow and rain)
suggestion concerning the unimarc import : I modified graphically .ui of the csv import and there is almost nothing to add (from my point of view)


put the catmandu engine in external parser (as a choice)
put the BiblioteQ engine in internal parser (as default choice)

and import like the csv with the choice of the columns

![suggestion-156](https://user-images.githubusercontent.com/56431039/147750613-97e87f72-bd75-4e4a-991d-1517aa2210ad.png)

[biblioteq_impor_unimarc-meteos77.ui.zip](https://github.com/textbrowser/biblioteq/files/7792826/biblioteq_impor_unimarc-meteos77.ui.zip)

the bash command I'm running :  catmandu convert MARC to CSV --field_sep "\n" --fix fix/unimarc_to_BiblioteQ_import_csv-v0.1.10.fix < Source_110000_fichier1-utf8.pan > BQ_110000-fix0.1.10.csv

**Comment by meteos77** [comment]
I revised my script from the beginning to be more independent from the name database fields.

the first part works with unimarc field names (200a_title for the 200 $a field of unimarc)

the second part of replacing values by names (lookup) is specific to French but can be translated (for example dewey codes -> dewey names)

the last part is the transformation of the variables to the BiblioteQ 200a_title -> title (I added 2 examples)

[2022-01-02_Unimarc_to_BiblioteQ_by_script_Catmandu-with_examples_2022-01-03_v1.zip](https://github.com/textbrowser/biblioteq/files/7801605/2022-01-02_Unimarc_to_BiblioteQ_by_script_Catmandu-with_examples_2022-01-03_v1.zip)

**Comment by meteos77** [comment]
suggestion : instead of having 2 almost similar interfaces (csv and unimarc), maybe create a general "csv or unimarc" import
BiblioteQ will make the right unimarc analysis before the csv import or the direct import if it is the csv

with the command file : 00_Source_110000_file1-utf8.pan: MARC21 Bibliographic
with the command file: file 03_Source_110000_fichier1-utf8.csv: CSV text

**Comment by meteos77** [comment]
I think we can close this conversation as it duplicates Super Magic Import #241.
I have created the tool in python that allows me to do what I want to transform a unimarc 995 into a csv to import it into BQ.
It works at 90%. (This allowed us to put BQ in our library)

I need to make more progress in python to be 100% happy with the tool.

If you want to make BQ unimarc file compatible, this feature will be welcome but I think it's a lot of work to get to the same point.

**I leave you alone to close.**

the analyzer:
[fonctionUniMarcFr.py.zip](https://github.com/textbrowser/biblioteq/files/11430464/fonctionUniMarcFr.py.zip)
